src/couch/src/couch_att.erl

```diff
@@ -732,13 +735,19 @@ upgrade_encoding(true) -> gzip;
 upgrade_encoding(false) -> identity;
 upgrade_encoding(Encoding) -> Encoding.
 
-
 max_attachment_size() ->
-    case config:get("couchdb", "max_attachment_size", "infinity") of
+    max_attachment_size(config:get("couchdb", "max_attachment_size", ?GB)).
+
+max_attachment_size(MaxAttSizeConfig) ->
+    case MaxAttSizeConfig of
         "infinity" ->
             infinity;
+        MaxAttSize when is_list(MaxAttSize) ->
+            list_to_integer(MaxAttSize);
+        MaxAttSize when is_integer(MaxAttSize) ->
+            MaxAttSize;
         MaxAttSize ->
-            list_to_integer(MaxAttSize)
+            erlang:error({invalid_max_attachment_size, MaxAttSize})
```

Comment:
If we are throwing new type of error we might need to define it here https://github.com/apache/couchdb/blob/main/src/chttpd/src/chttpd.erl#L930 and maybe here https://github.com/apache/couchdb/blob/main/src/chttpd/src/chttpd.erl#L335

Reply:
Good catch! What do you think of 921aa502e instead though?